gradio/routes.py

```diff
@@ -145,12 +145,16 @@ def get_blocks(self) -> gradio.Blocks:
             raise ValueError("No Blocks has been configured for this app.")
         return self.blocks
 
-    @staticmethod
-    def build_proxy_request(url_path):
+    def build_proxy_request(self, url_path):
         url = httpx.URL(url_path)
-        is_hf_url = url.host.endswith(".hf.space")
+        assert self.blocks
+        # Don't proxy a URL unless it's a URL specifically loaded by the user using
+        # gr.load() to prevent SSRF or harvesting of HF tokens by malicious Spaces.
+        is_safe_url = any(url.host.endswith(root) for root in self.blocks.root_urls)
```

Comment:
If `self.blocks.root_urls` is `['foo.hf.space']` this will consider `'barfoo.hf.space'` safe. I don't think that's what you meant here?

Reply:
Ah right, will fix

Reply:
Fixed now